public_html/wp-content/plugins/camptix/addons/require-login.php

```diff
@@ -936,6 +940,79 @@ public function hide_unconfirmed_attendees( $query_args ) {
 		return $query_args;
 	}
 
+	/**
+	 * Add the ability to filter camptix notify segments by confirmed status.
+	 *
+	 * @param array $segments
+	 * @return array
+	 */
+	public function camptix_notify_segment_fields( $segments ) {
+		$segments[] = [
+			'caption'      => 'Ticket Status',
+			'option_value' => 'ticket_status',
+			'type'         => 'select',
+			'ops'          => [ 'is' ],
+			'values'       => [
+				[
+					'caption' => __( 'Confirmed', 'wordcamporg' ),
+					'value'   => 'confirmed',
+				],
+				[
+					'caption' => __( 'Unconfirmed', 'wordcamporg' ),
+					'value'   => 'unconfirmed',
+				],
+			],
+		];
+
+		return $segments;
+	}
+
+	/**
+	 * Add the ability to filter camptix notify segments by confirmed status.
+	 *
+	 * @param array $query      The posts query arguments.
+	 * @param array $conditions The conditions to filter by.
+	 * @return array
+	 */
+	public function camptix_notify_segment_query( $query, $conditions ) {
+		foreach ( $conditions as $condition ) {
+			if ( 'ticket_status' === $condition['field'] ) {
+				if ( 'unconfirmed' === $condition['value'] ) {
+					$query['meta_query'][] = array(
+						// All of these are unconfirmed.
+						'relation' => 'or',
+						// Unconfirmed username listed.
+						array(
+							'key' => 'tix_username',
+							'value' => self::UNCONFIRMED_USERNAME,
+							'compare' => '=',
+						),
+						// Has no username linked.
+						array(
+							'key' => 'tix_username',
+							'compare' => 'NOT EXISTS',
+						),
+						// The email is the standard unknown attendee.
+						array(
+							'key' => 'tix_email',
+							'value' => $this->get_unknown_attendee_info()['email'],
+							'compare' => '=',
+						),
```

Comment:
If the email is the unknown value, the message sent by notify won't go anywhere, so do we need this part of the query?